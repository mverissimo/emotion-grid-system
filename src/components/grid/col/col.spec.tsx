import { render } from '@testing-library/react';

import { Col } from '.';

describe('@Grid/Col', () => {
  it('should render with default styles', () => {
    const { container } = render(<Col>Col</Col>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with size styles', () => {
    const { container } = render(
      <Col
        size={{
          initial: 1,
          sm: 2,
          md: 3,
          lg: 4,
          xl: 5,
        }}
      >
        Col
      </Col>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with offset styles', () => {
    const { container } = render(
      <Col
        offset={{
          initial: 1,
          sm: 2,
          md: 3,
          lg: 4,
          xl: 5,
        }}
      >
        Col
      </Col>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with a custom className', () => {
    const { container } = render(
      <Col className="my-custom-class-name">Col</Col>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
