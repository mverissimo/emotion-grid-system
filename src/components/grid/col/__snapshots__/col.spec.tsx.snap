// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@Grid/Col > should render with a custom className 1`] = `
.emotion-0 {
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  box-sizing: border-box;
  width: 100%;
  max-width: 100%;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
}

@media (min-width: 40rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 48rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 64rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 96rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

<div
  class="emotion-0 my-custom-class-name"
  data-grid="col"
>
  Col
</div>
`;

exports[`@Grid/Col > should render with default styles 1`] = `
.emotion-0 {
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  box-sizing: border-box;
  width: 100%;
  max-width: 100%;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
}

@media (min-width: 40rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 48rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 64rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 96rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

<div
  class="emotion-0"
  data-grid="col"
>
  Col
</div>
`;

exports[`@Grid/Col > should render with offset styles 1`] = `
.emotion-0 {
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  box-sizing: border-box;
  width: 100%;
  max-width: 100%;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
  margin-left: calc((1 / 4) * 100%);
}

@media (min-width: 40rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 48rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 64rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 96rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 40rem) {
  .emotion-0 {
    margin-left: calc((2 / 8) * 100%);
  }
}

@media (min-width: 48rem) {
  .emotion-0 {
    margin-left: calc((3 / 12) * 100%);
  }
}

@media (min-width: 64rem) {
  .emotion-0 {
    margin-left: calc((4 / 12) * 100%);
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: calc((5 / 12) * 100%);
  }
}

<div
  class="emotion-0"
  data-grid="col"
>
  Col
</div>
`;

exports[`@Grid/Col > should render with size styles 1`] = `
.emotion-0 {
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  box-sizing: border-box;
  width: 100%;
  max-width: 100%;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
  -webkit-flex: 0 0 calc((1 / 4) * 100%);
  -ms-flex: 0 0 calc((1 / 4) * 100%);
  flex: 0 0 calc((1 / 4) * 100%);
}

@media (min-width: 40rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 48rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 64rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 96rem) {
  .emotion-0 {
    padding-left: calc(2rem / 2);
    padding-right: calc(2rem / 2);
  }
}

@media (min-width: 40rem) {
  .emotion-0 {
    -webkit-flex: 0 0 calc((2 / 8) * 100%);
    -ms-flex: 0 0 calc((2 / 8) * 100%);
    flex: 0 0 calc((2 / 8) * 100%);
  }
}

@media (min-width: 48rem) {
  .emotion-0 {
    -webkit-flex: 0 0 calc((3 / 12) * 100%);
    -ms-flex: 0 0 calc((3 / 12) * 100%);
    flex: 0 0 calc((3 / 12) * 100%);
  }
}

@media (min-width: 64rem) {
  .emotion-0 {
    -webkit-flex: 0 0 calc((4 / 12) * 100%);
    -ms-flex: 0 0 calc((4 / 12) * 100%);
    flex: 0 0 calc((4 / 12) * 100%);
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    -webkit-flex: 0 0 calc((5 / 12) * 100%);
    -ms-flex: 0 0 calc((5 / 12) * 100%);
    flex: 0 0 calc((5 / 12) * 100%);
  }
}

<div
  class="emotion-0"
  data-grid="col"
>
  Col
</div>
`;
